src/bindings/js/node/src/infer_request.cpp

```diff
@@ -190,3 +197,82 @@ void InferRequestWrap::infer(const Napi::Object& inputs) {
 Napi::Value InferRequestWrap::get_compiled_model(const Napi::CallbackInfo& info) {
     return CompiledModelWrap::Wrap(info.Env(), _infer_request.get_compiled_model());
 }
+
+struct TsfnContext {
+    TsfnContext(Napi::Env env) : deferred(Napi::Promise::Deferred::New(env)){};
+
+    Napi::Promise::Deferred deferred;
+    ov::InferRequest* _context_ir;
+    std::thread nativeThread;
+
+    Napi::ThreadSafeFunction tsfn;
+    std::vector<ov::Tensor> input_tensors;
+    std::map<std::string, ov::Tensor> result;
+};
+
+void FinalizerCallback(Napi::Env env, void* finalizeData, TsfnContext* context) {
+    context->nativeThread.join();
+    delete context;
+};
+
+void performInferenceThread(TsfnContext* data) {
+    infer_mutex.lock();
+    for (size_t i = 0; i < data->input_tensors.size(); ++i) {
+        data->_context_ir->set_input_tensor(i, data->input_tensors[i]);
+    }
+    data->_context_ir->infer();
+
+    auto compiled_model = data->_context_ir->get_compiled_model().outputs();
+    std::map<std::string, ov::Tensor> outputs;
+
+    for (auto& node : compiled_model) {
+        auto tensor = data->_context_ir->get_tensor(node);
+        auto new_tensor = ov::Tensor(tensor.get_element_type(), tensor.get_shape());
+        tensor.copy_to(new_tensor);
```

Comment:
```suggestion
        std::memcpy(new_tensor.data(), tensor.data(), new_tensor.get_byte_size());
```
`tensor.copy_to` is complex and handle some different layouts etc. So if `memcpy` will work use it as much simpler than `copy_to`